### DISCUSSION TITLE
Quick question: io.opentelemetry.api.trace.Span can be inside @ApplicationScoped bean?

### DISCUSSION THREAD

**Discussion by thomaswiradikusuma** [quarkusio/quarkus#41652] [Q&A] [answered]
As per subject, because we can do `span.setAttribute("foo", foo)`, but https://javadoc.io/doc/io.opentelemetry/opentelemetry-api/latest/io/opentelemetry/api/trace/Span.html says it's `@ThreadSafe`.

[a comment by an automated account is not shown]

**Reply by brunobat** [reply, marked as answer]
Yes it can, however, you should only set attributes that make sense for an application scoped bean.
As an example, if you want to have access, for example to the user id (different in every request), it needs to be on a `@RequestScoped` bean.
